demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+                                             sampleazureiotCONNACK_RECV_TIMEOUT_MS );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        xResult = AzureIoTHubClient_SubscribeCloudToDeviceMessage( &xAzureIoTHubClient, prvHandleCloudMessage,
+                                                                   &xAzureIoTHubClient, sampleazureiotWAIT_FOREVER );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        xResult = AzureIoTHubClient_SubscribeDirectMethod( &xAzureIoTHubClient, prvHandleDirectMethod,
+                                                           &xAzureIoTHubClient, sampleazureiotWAIT_FOREVER );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        xResult = AzureIoTHubClient_SubscribeDeviceTwin( &xAzureIoTHubClient, prvHandleDeviceTwinMessage,
+                                                         &xAzureIoTHubClient, sampleazureiotWAIT_FOREVER );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        /* Get the device twin on boot */
+        xResult = AzureIoTHubClient_GetDeviceTwin( &xAzureIoTHubClient );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        /* Create a bag of properties for the telemetry */
+        xResult = AzureIoT_MessagePropertiesInit( &xPropertyBag, ucPropertyBuffer, 0, sizeof( xPropertyBag ) );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        xResult = AzureIoT_MessagePropertiesAppend( &xPropertyBag, ( uint8_t * ) "name", sizeof( "name" ) - 1,
+                                                    ( uint8_t * ) "value", sizeof( "value" ) - 1 );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        /* Publish messages with QoS1, send and process Keep alive messages. */
+        for( ulPublishCount = 0; ulPublishCount < ulMaxPublishCount; ulPublishCount++ )
+        {
+            xResult = AzureIoTHubClient_SendTelemetry( &xAzureIoTHubClient,
+                                                       ( const uint8_t * ) sampleazureiotMESSAGE,
+                                                       sizeof( sampleazureiotMESSAGE ) - 1,
+                                                       &xPropertyBag );
+            configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+            LogInfo( ( "Attempt to receive publish message from IoTHub.\r\n" ) );
+            xResult = AzureIoTHubClient_ProcessLoop( &xAzureIoTHubClient,
+                                                     sampleazureiotPROCESS_LOOP_TIMEOUT_MS );
+            configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+            if( ulPublishCount % 2 == 0 )
+            {
+                /* Send reported property every other cycle */
+                xResult = AzureIoTHubClient_SendDeviceTwinReported( &xAzureIoTHubClient,
+                                                                    ( const uint8_t * ) sampleazureiotTWIN_PROPERTY,
```

Comment:
Consider on this one doing something like 
{

   char sampleazureiotTWIN_PROPERTY[SomeBufferSize];
   sprintf(sampleazureiotTWIN_PROPERTY, "TwinIterationForCurrentConnection:%d", ++someGlobalCounter)"
}

Check that sprintf() is cool on FreeRTOS or potentiall they have a stripped down one, right, that could handle a %d.

The reason I recommend this is that repeatedl telemetry sends I can easily see in IoT Explorer, even if of same data.  But I don't get the sense that the data is being updated by looking at a DeviceTwin if we always send the same thing.